internal/generator/fluentd/output/cloudwatch/cloudwatch.go

```diff
@@ -106,7 +107,9 @@ func SecurityConfig(o logging.OutputSpec, secret *corev1.Secret) Element {
 	// First check for credentials or role_arn key, indicating a sts-enabled authentication
 	if security.HasAwsRoleArnKey(secret) || security.HasAwsCredentialsKey(secret) {
 		return AWSKey{
-			KeyRoleArn: ParseRoleArn(secret),
+			KeyRoleArn:          ParseRoleArn(secret),
+			KeyRoleSessionName:  constants.AWSRoleSessionName,
+			KeyWebIdentityToken: path.Join(constants.AWSWebIdentityTokenMount, constants.AWSWebIdentityTokenFilePath),
```

Comment:
Maybe I don't understand, but this identity token is constant across the configured outputs but its the 'arn' which is variable?

Reply:
Correct, the SA token is shared among all outputs, since it is not tied the role.   The latest/rotated token is used to obtain temporary credentials for each "role request", allowing the SA to assume that role during the request session.